packages/headless/src/controllers/core/folded-result-list/headless-core-folded-result-list.ts

```diff
@@ -0,0 +1,277 @@
+   *
+   * @param result - The result to send analytics for.
+   */
+  logShowMoreFoldedResults(result: Result): void;
+  /**
+   * Logs a custom event when a user shows less folded results.
+   *   */
+  logShowLessFoldedResults(): void;
+  /**
+   * Finds a folded result by its unique ID.
+   *
+   * @param collection - The folded collection whose ID will be used to find a collection in the results.
+   * @returns The `FoldedResult` associated with the collection's ID.
+   */
+  findResultById(collection: FoldedCollection): FoldedResult | null;
+  /**
+   * Finds a folded result by its collection.
+   *
+   * @param collection - The folded collection whose collection name will be used to find a collection in the results.
+   * @returns The `FoldedResult` associated with the collection's name.
+   */
+  findResultByCollection(collection: FoldedCollection): FoldedResult | null;
+}
+
+/**
+ * A scoped and simplified part of the headless state that is relevant to the `FoldedResultList` controller.
+ */
+export interface CoreFoldedResultListState extends SearchStatusState {
+  /**
+   * The ordered list of collections.
+   * */
+  results: FoldedCollection[];
+  /**
+   * The unique identifier of the response where the results were fetched, this value does not change when loading more results.
+   */
+  searchResponseId: string;
+  /**
+   * Whether more results are available, using the same parameters as the last successful query.
+   *
+   * This property is not compatible with the `Pager` controller.
+   */
+  moreResultsAvailable: boolean;
+}
+
+export interface FoldedResultAnalyticsClient {
+  logShowMoreFoldedResults: (result: Result) => ClickAction;
+  logShowLessFoldedResults: () => CustomAction;
+}
+
+/**
+ * Creates a core `FoldedResultList` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `FoldedResultList` properties.
+ * @param analyticsClient - A FoldedResultAnalyticsClient to send the appropriate analytics calls.
+ * @returns A `CoreFoldedResultList` controller instance.
+ */
+export function buildCoreFoldedResultList(
+  engine: CoreEngine,
+  props: CoreFoldedResultListProps,
+  analyticsClient: FoldedResultAnalyticsClient
+): CoreFoldedResultList {
+  if (!loadFoldingReducer(engine)) {
+    throw loadReducerError;
+  }
+
+  const controller = buildCoreResultList(engine, props);
+  const {dispatch} = engine;
+  const getState = () => engine.state;
+
+  const options = props?.options?.folding
+    ? validateOptions(
+        engine,
+        optionsSchema,
+        props.options!.folding!,
+        'buildFoldedResultList'
+      )
+    : {};
+
+  dispatch(registerFolding({...options}));
+
+  return {
+    ...controller,
+
+    loadCollection: (collection) => {
+      dispatch(
+        props.loadCollectionActionCreator(
+          collection?.result?.raw[
+            engine.state.folding.fields.collection
+          ] as string
```

Comment:
Same as above. I don't see why this optional chain is needed, since `collection` and `result` both seem required.

```suggestion
          collection.result.raw[
            engine.state.folding.fields.collection
          ] as string
```